pyfr/integrators/dual/phys/steppers.py

```diff
@@ -1,25 +1,125 @@
 # -*- coding: utf-8 -*-
 
+from itertools import chain
+
 from pyfr.integrators.dual.phys.base import BaseDualIntegrator
 
 
 class BaseDualStepper(BaseDualIntegrator):
-    pass
 
+    def _set_currstg(self, s):
+        self.currstg = s
+
+    def _finalize_stage(self, tcurr):
+        pass
+
+
+class BaseBDFStepper(BaseDualStepper):
+    _nstages = 1
+
+    @property
+    def stage_nregs(self):
```

Comment:
No need to make these properties, just assign directly as we do in the regular steppers.

Reply:
I tried assigning this in the new init function, but it doesn't work like that. We pass stepper_nregs and stage_nregs down at few different locations, and it has to be defined either as a function or set to a value before we initialize. We don't know how many stage_nregs we need before we know the stepper